packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -1078,8 +1082,7 @@ export class Table extends TableBase {
       public readonly tableArn: string;
       public readonly tableStreamArn?: string;
       public readonly encryptionKey?: kms.IKey;
-      protected readonly hasIndex = (attrs.globalIndexes ?? []).length > 0 ||
-        (attrs.localIndexes ?? []).length > 0;
+      protected readonly hasIndex = true;
```

Comment:
Instead of keeping this property and changing it to `true` can we just remove all usage of it (since
it's always true now)?

Reply:
Thank you for the review! I was wondering if there was a use case to grant granular permissions with the Table class. To grant permissions for indexed with imported table, users need to additionally provide index information. However, with the original Table class, users are supposed to provide index information to create indexes. Then the minimum permissions are granted. The behavior seems better since ideally only permissions users need should be given.
If there is no concern to remove this behavior, I'll remove `hasIndex` property from the base class.

Reply:
I like the idea to add a new parameter. I would go with `forceGrantIndexPermissions?` to make it less confusing when we provide both parameters.

> Currently it looks like on both imported tables and managed tables as long as there is any index we just grant access to all indexes. I think it would be better if we granted access to the specific indexes if they are provided so I want to leave that option open for a future PR (it would need a feature flag).

Agreed. It would be also nice that grant method accepts a parameter to specify how the permissions are granted (e.g. all indexes, specific indexes, no index) in the future.

I'll update the PR. Thanks!

Reply:
On second thought, if we want to leave the possibility for the future extensions you mentioned, `grantIndexPermissions` seems better. Please take a look.